src/App.tsx

```diff
@@ -1,26 +1,197 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { FormEvent, useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { addTodo, getTodos, removeTodo, USER_ID } from './api/todos';
+import { Todo, TodoData } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { ErrorMessage, FilterStatus } from './enums/enums';
+import { TodoItem } from './components/TodoItem/TodoItem';
 
-const USER_ID = 0;
+const filterTodos = (todos: Todo[], status: FilterStatus) => {
+  let filteredTodos = [...todos];
+
+  if (status === FilterStatus.Active) {
+    filteredTodos = filteredTodos.filter(todo => !todo.completed);
+  }
+
+  if (status === FilterStatus.Completed) {
+    filteredTodos = filteredTodos.filter(todo => todo.completed);
+  }
+
+  return filteredTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [statusValue, setStatusValue] = useState(FilterStatus.All);
+  const [error, setError] = useState('');
+  const [todoTitleValue, setTodoTitleValue] = useState('');
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const [todosDeleting, setTodosDeleting] = useState<Todo['id'][]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const todosCompleted = useMemo(
+    () => todos.filter(todo => todo.completed).length,
+    [todos],
+  );
+
+  const todosActive = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const showErrorContainer = (errorMessage: string) => {
+    setError(errorMessage);
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    async function fetchData() {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
    const fetchData = async () => {
```